mock/py/mockbuild/backend.py

```diff
@@ -567,9 +636,21 @@ def buildsrpm(self, spec, sources, timeout, follow_links):
                 else:
                     shutil.copy(sources, self.buildroot.make_chroot_path(self.buildroot.builddir, "SOURCES"))
 
-            spec = self.buildroot.make_chroot_path(self.buildroot.builddir, "SPECS", os.path.basename(spec))
+                host_chrootsources = self.buildroot.make_chroot_path(self.buildroot.builddir, "SOURCES")
```

Comment:
Having "chrootsources" in hand, you could pretty easily call something like
`call_hooks("preprocess_sources", chrootsources)`.  That way it would be much
easier to add any other source preprocessors.

Reply:
Right, but I think I would also need access to the specfile parameter (which was specified by `--spec`) and also access to the chroot environment. I guess this wouldn't be a problem to get to those variables as well, however.

Reply:
I mean access to the environment unless you will want me to switch to evaluation on host :).

Reply:
AFAIK plugins have everything important in hand, namely reference to buildroot object.